src/main/java/Weng.java

```diff
@@ -0,0 +1,181 @@
+import Exceptions.DescriptionEmptyException;
+import Exceptions.IllegalCommandException;
+import Exceptions.MissingDatesException;
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+import static java.lang.Integer.parseInt;
+
+public class Weng {
+    public static final String SEPARATOR = "____________________________________________________________";
+    private static Task[] tasks = new Task[100];
+    private static int totalNumTasks = 0;
+
+    public static Task[] getTasks() {
+        return tasks;
+    }
+
+    public static void setTasks(Task[] tasks) {
+        Weng.tasks = tasks;
+    }
+
+    public static void addTodo(String[] currLine) {
+        String description = String.join(" ", Arrays.copyOfRange(currLine, 1, currLine.length));
+        addAndPrintHandler(new Todo(description));
+    }
+
+    public static void addEvent(String[] input) throws MissingDatesException {
+        addAndPrintHandler(new Event(input));
+    }
+
+    public static void addDeadline(String[] input) throws MissingDatesException {
+        addAndPrintHandler(new Deadline(input));
+    }
+
+    public static void setIndexInTask(Task task, int index) {
+        Weng.tasks[index] = task;
+    }
+
+    public static int getTotalNumTasks() {
+        return totalNumTasks;
+    }
+
+    public static void setTotalNumTasks(int totalNumTasks) {
+        Weng.totalNumTasks = totalNumTasks;
+    }
+
+    public static void main(String[] args) {
+        printGreeting();
+        inputVerification();
+    }
+
+    private static void inputVerification() {
+        Scanner scanner = new Scanner(System.in);
+        String line;
+        //LEGAL_COMMANDS only used in this method, so ill leave it here
+        String[] LEGAL_COMMANDS = {"bye", "list", "todo", "deadline", "event", "unmark", "mark"};
+        boolean looping = true;
+        while (looping) {
+            try {
+                //hmm, not sure how to refactor this code effectively
+                //do give comments if got any idea
+                line = scanner.nextLine();
+                String[] currLine = line.split(" ");
+                printSeparator();
+
+                if (Arrays.stream(LEGAL_COMMANDS).noneMatch(currLine[0]::equals)) {
+                    //if the command is not in the list of legal commands
+                    throw new IllegalCommandException();
+                }
+
+                switch (currLine[0]) {
+                case "bye":
+                    printGoodbye();
+                    looping = false; //break loop
```

Comment:
Some of the comments are describing an obvious line of code.
We will discuss this in the near future.